mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxMessageDAOUtils.java

```diff
@@ -181,4 +185,10 @@ public long size() {
                 throw new RuntimeException("Unknown FetchType " + fetchType);
         }
     };
+
+    static Mono<PostgresMailboxMessageDAO> mailboxMessageDAO(PostgresExecutor.Factory executorFactory, MailboxSession session) {
```

Comment:
What's the issue exactly? Looks overkill to me without a bit of details

Reply:
We have 6 classes 
```
AllSearchOverride.java
DeletedSearchOverride.java
DeletedWithRangeSearchOverride.java
NotDeletedWithRangeSearchOverride.java
UidSearchOverride.java
UnseenSearchOverride.java
```

And each class, we have a code
```
 Mono.just(session.getUser().getDomainPart())
            .map(executorFactory::create)
            .map(PostgresMailboxMessageDAO::new)
            .flatmap(.....
```

Reply:
You mean James is doing that for each override when starting correct?

Ok but can't we have a more elegant way here?